File: packages/android_intent_plus/lib/src/put_classes/put_float.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutFloat extends PutBase<double> {
  PutFloat({required String key, required double value})
      : super(key: key, value: value);

  @override
  String get javaClass => 'PutFloat';

  factory PutFloat.fromJson({required String key, required dynamic value}) {
    return PutFloat(key: key, value: value as double);
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutFloat.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;

import org.json.JSONException;
import org.json.JSONObject;

import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;


public class PutFloat extends PutBase<Float> {

  public static final String JAVA_CLASS = "PutFloat";

  public PutFloat(String key, Float value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutFloat putDouble) {
    bundle.putFloat(putDouble.key, putDouble.value);
  }

  public static PutFloat fromJson(JSONObject jsonObject) throws JSONException {
    return new PutFloat(jsonObject.getString(Constants.KEY), (float) jsonObject.getDouble(Constants.VALUE));
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", value);
    return jsonObject;
  }
}
